geomstats/information_geometry/fisher_rao_metric.py

```diff
@@ -0,0 +1,157 @@
+
+
+class FisherRaoMetric(RiemannianMetric):
+    r"""Class to derive the information metric from the pdf in InformationManifold.
+
+    Given a statistical manifold with coordinates :math:`\theta`,
+    the Fisher information metric is:
+    :math:`g_{j k}(\theta)=\int_X \frac{\partial \log p(x, \theta)}{\partial \theta_j}
+        \frac{\partial \log p(x, \theta)}{\partial \theta_k} p(x, \theta) d x`
+
+    Attributes
+    ----------
+    information_manifold : InformationManifold object
+        Riemannian Manifold for a parametric family of (real) distributions.
+    support : list, shape = (2,)
+        Left and right bounds for the support of the distribution.
+        But this is just to help integration, bounds should be as large as needed.
+    """
+
+    def __init__(self, information_manifold, support):
+        dim = information_manifold.dim
+        super().__init__(dim=dim, shape=(dim,), signature=(dim, 0))
+        self.information_manifold = information_manifold
+        self.support = support
+
+    def metric_matrix(self, base_point):
+        r"""Compute the inner-product matrix.
+
+        The Fisher information matrix is noted I in the literature.
+
+        Compute the inner-product matrix of the Fisher information metric
+        at the tangent space at base point.
+
+        :math: `I(\theta) = \mathbb{E}[(\partial_{\theta}(\log(pdf_{\theta}(x))))^2]`
+
+        Parameters
+        ----------
+        base_point : array-like, shape=[..., dim]
+            Base point.
+
+        Returns
+        -------
+        mat : array-like, shape=[..., dim, dim]
+            Inner-product matrix.
+
+        References
+        ----------
+        .. [AS1985] Amari, S (1985)
+            Differential Geometric Methods in Statistics, Berlin, Springer – Verlag.
+        """
+
+        def log_pdf_at_x(x):
+            """Compute the log function of the pdf for a fixed value on the support.
+
+            Parameters
+            ----------
+            x : float, shape (,)
+                Point on the support of the distribution
+            """
+            return lambda point: gs.log(
+                self.information_manifold.point_to_pdf(point)(x)
+            )
+
+        def dlog(x):
+            """Compute the derivative of the log-pdf with respect to the parameters.
+
+            Parameters
+            ----------
+            x : float, shape (,)
+                Point on the support of the distribution
+            """
+            return gs.autodiff.jacobian(log_pdf_at_x(x))(base_point)
+
+        def squared_dlog(x):
+            """Compute the square (in matrix terms) of dlog.
+
+            This is the variable whose expectance is the Fisher-Rao information.
+
+            Parameters
+            ----------
+            x : float, shape (,)
+                Point on the support of the distribution
+            """
+            return gs.einsum("...i, ...j -> ...ij", dlog(x), dlog(x))
+
+        metric_mat = quad_vec(
```

Comment:
Does this work well with the different backends? e.g. the output is `torch.Tensor` if `torch` backend is used? (looks unlikely.)

I think we may have to consider soon to have a more controlled access to scipy. I have some ideas, we can discuss them later.

Reply:
It worked in all backends in the tests from fisher_rao_metric.py
I think it converts the tensors to numpy.
However, it will not to "detach" thus we might get bug there.

Agree that we should think about this more very curious to hear your thoughts on this!